syntax = "proto3";

package flask_api;

// User service
service UserService {
    rpc GetUser (GetUserRequest) returns (UserResponse);
    rpc ListUsers (ListUsersRequest) returns (ListUsersResponse);
    rpc CreateUser (CreateUserRequest) returns (UserResponse);
}

// Task service
service TaskService {
    rpc GetTask (GetTaskRequest) returns (TaskResponse);
    rpc ListTasks (ListTasksRequest) returns (ListTasksResponse);
    rpc CreateTask (CreateTaskRequest) returns (TaskResponse);
}

// User messages
message User {
    int32 id = 1;
    string username = 2;
    string email = 3;
    string created_at = 4;
}

message GetUserRequest {
    int32 id = 1;
}

message ListUsersRequest {
    int32 page = 1;
    int32 page_size = 2;
}

message CreateUserRequest {
    string username = 1;
    string email = 2;
}

message UserResponse {
    User user = 1;
}

message ListUsersResponse {
    repeated User users = 1;
}

// Task messages
message Task {
    int32 id = 1;
    string title = 2;
    string description = 3;
    bool completed = 4;
    int32 user_id = 5;
    string created_at = 6;
}

message GetTaskRequest {
    int32 id = 1;
}

message ListTasksRequest {
    int32 page = 1;
    int32 page_size = 2;
}

message CreateTaskRequest {
    string title = 1;
    string description = 2;
    bool completed = 3;
    int32 user_id = 4;
}

message TaskResponse {
    Task task = 1;
}

message ListTasksResponse {
    repeated Task tasks = 1;
}
